docs/how-to-guides/block-tutorial/block-supports-in-dynamic-blocks.md

```diff
@@ -0,0 +1,217 @@
+	}
+	if ( ! empty( $block_attributes['textColor'] ) ) {
+		$styles .= "color:{$block_attributes['textColor']};";
+	}
+	$wrapper_attributes = get_block_wrapper_attributes();
+	return sprintf(
+		'<h3 %1$s href="%2$s" style="%3$s">%4$s<h3>',
+		$wrapper_attributes,
+		esc_url( get_permalink( $post_id ) ),
+		esc_attr( $styles ),
+		esc_html( get_the_title( $post_id ) )
+	);
+}
+
+function gutenberg_examples_dynamic() {
+	register_block_type(
+		'gutenberg-examples/example-dynamic',
+		array(
+			'api_version'       => 2,
+			'category'          => 'widgets',
+			'attributes'        => array(
+				'bgColor'   => array( 'type' => 'string' ),
+				'textColor' => array( 'type' => 'string' ),
+			),
+			'render_callback'   => 'gutenberg_examples_dynamic_render_callback',
+			'skip_inner_blocks' => true,
+		)
+	);
+}
+add_action( 'init', 'gutenberg_examples_dynamic' );
+
+```
+
+### With block supports
+
+Let's see how we can achieve the same functionality, but by using `block supports`.
+
+```jsx
+import { registerBlockType } from '@wordpress/blocks';
+import { useSelect } from '@wordpress/data';
+import { useBlockProps } from '@wordpress/block-editor';
+
+registerBlockType( 'gutenberg-examples/example-dynamic-block-supports', {
+	apiVersion: 2,
+	title: 'Example: last post title(block supports)',
+	icon: 'megaphone',
+	category: 'widgets',
+	edit: function BlockEdit() {
+		const blockProps = useBlockProps();
+		const posts = useSelect( ( select ) => {
+			return select( 'core' ).getEntityRecords( 'postType', 'post', {
+				per_page: 1,
+			} );
+		}, [] );
+		if ( ! posts ) {
+			return null;
+		}
+		return (
+			<div { ...blockProps }>
+				{ !! posts.length && <h3>{ posts[ 0 ].title.rendered }</h3> }
+			</div>
+		);
+	},
+	supports: { color: true },
+} );
+```
+
+And the server side part becomes:
+
+```php
+<?php
+function gutenberg_examples_dynamic_block_supports_render_callback( $block_attributes, $content ) {
+	$recent_posts = wp_get_recent_posts( array(
+		'numberposts' => 1,
+		'post_status' => 'publish',
+	) );
+	if ( count( $recent_posts ) === 0 ) {
+		return 'No posts';
+	}
+	$post    = $recent_posts[ 0 ];
+	$post_id = $post['ID'];
+	$wrapper_attributes = get_block_wrapper_attributes();
+	return sprintf(
+		'<h3 %1$s href="%2$s">%3$s<h3>',
+		$wrapper_attributes,
+		esc_url( get_permalink( $post_id ) ),
+		esc_html( get_the_title( $post_id ) )
+	);
+}
+
+function gutenberg_examples_dynamic_block_supports() {
+	register_block_type(
+		'gutenberg-examples/example-dynamic-block-supports',
+		array(
+			'api_version'       => 2,
+			'category'          => 'widgets',
+			'supports'          => array( 'color' => true ),
+			'render_callback'   => 'gutenberg_examples_dynamic_block_supports_render_callback',
+			'skip_inner_blocks' => true,
+		)
+	);
+}
+add_action( 'init', 'gutenberg_examples_dynamic_block_supports' );
+
+```
+
+And that's it, the addition of the "supports" key above, will automatically make the following changes to the block:
```

Comment:
This part is copied from https://github.com/WordPress/gutenberg/pull/38210. I think these two should be identical, so any change should be reflected to both tutorials.